server/tserver/src/main/java/org/apache/accumulo/tserver/scan/LookupTask.java

```diff
@@ -79,6 +79,9 @@ public void run() {
       long maxScanTime = 4000;
 
       long startTime = System.currentTimeMillis();
+      LinkedList<Pair<KeyExtent,List<Range>>> queryQueue = new LinkedList<>();
+      session.queries.entrySet()
+          .forEach(e -> queryQueue.addLast(new Pair<>(e.getKey(), e.getValue())));
```

Comment:
We need to make sure the session.queries is reset with what is left in the queryQueue after the loop.  Otherwise whenever the scan session is restarted (e.g. bytesAdded > maxResultsSize) then we need to restart where we left off.

Reply:
Committed a change to remove and add to session.queries whenever a change is made to queryQueue.